modules/extensions/src/main/java/org/apache/synapse/mediators/bsf/CommonScriptMessageContext.java

```diff
@@ -0,0 +1,848 @@
+     * @throws ScriptException For errors in converting xml To OM
+     * @throws OMException     For errors in OM manipulation
+     */
+
+    @Override
+    public void setPayloadXML(Object payload) throws OMException, ScriptException {
+        SOAPBody body = mc.getEnvelope().getBody();
+        OMElement firstChild = body.getFirstElement();
+        OMElement omElement = xmlHelper.toOMElement(payload);
+        if (firstChild == null) {
+            body.addChild(omElement);
+        } else {
+            firstChild.insertSiblingAfter(omElement);
+            firstChild.detach();
+        }
+    }
+
+    /**
+     * Get the JSON object representation of the JSON message body of the request.
+     *
+     * @return JSON object of the message body
+     */
+    @Override
+    public Object getPayloadJSON() {
+        return jsonObject(mc);
+    }
+
+    @Override
+    public Object getJsonText() {
+        if (mc == null) {
+            return "";
+        }
+        Object text = mc.getProperty(JSON_TEXT);
+        return text == null ? "{}" : text;
+    }
+
+    /**
+     * Get the Message Payload as a text.
+     *
+     * @return Payload as text
+     */
+    @Override
+    public String getPayloadText() {
+        if (JsonUtil.hasAJsonPayload(((Axis2MessageContext) mc).getAxis2MessageContext())) {
+            return JsonUtil.jsonPayloadToString(((Axis2MessageContext) mc).getAxis2MessageContext());
+        } else {
+            return mc.getEnvelope().toString();
+        }
+    }
+
+    /**
+
+    /**
+     * Saves the payload of this message context as a JSON payload.
+     *
+     * @param jsonPayload Javascript native object to be set as the message body
+     * @throws ScriptException in case of creating a JSON object out of
+     *                         the javascript native object.
+     */
+    public void setPayloadJSON0(Object jsonPayload) throws ScriptException {
+        org.apache.axis2.context.MessageContext messageContext;
+        messageContext = ((Axis2MessageContext) mc).getAxis2MessageContext();
+
+        String jsonString;
+        if (jsonPayload instanceof String) {
+            jsonString = (String) jsonPayload;
+        } else {
+            jsonString = serializeJSON(jsonPayload);
+        }
+        try {
+            JsonUtil.getNewJsonPayload(messageContext, jsonString, true, true);
+        } catch (AxisFault axisFault) {
+            throw new ScriptException(axisFault);
+        }
+        //JsonUtil.setContentType(messageContext);
+        Object jsonObject = scriptEngine.eval('(' + jsonString + ')');
+        setJsonObject(mc, jsonObject);
+    }
+
+    /**
+     * Saves the JavaScript Object to the message context.
+     * @param messageContext
+     * @param jsonObject
+     * @return
+     */
+    @Override
+    public boolean setJsonObject(MessageContext messageContext, Object jsonObject) {
+        if (jsonObject == null) {
+            logger.error("Setting null JSON object.");
```

Comment:
If user need to set null to JSON_OBJECT, then this error logs every time he does that does. That will not work in the production environment. Isn't it intended usage?